cmd/set/puppet.go

```diff
@@ -0,0 +1,46 @@
+package set
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver"
+	"github.com/puppetlabs/prm/pkg/prm"
+	"github.com/spf13/cobra"
+)
+
+type setPuppetCommand struct {
+	Config prm.ConfigI
+}
+
+var puppetSemVer *semver.Version
+
+func (spc *setPuppetCommand) createSetPuppetCommand() *cobra.Command {
+	tmp := &cobra.Command{
+		Use:     "puppet <VERSION>",
+		Short:   "Sets the Puppet runtime to the specified version",
+		Long:    `Sets the Puppet runtime to the specified version`,
+		PreRunE: spc.preExecute,
+		Run:     spc.setPuppetVersion,
+	}
+
+	return tmp
+}
+
+func (spc *setPuppetCommand) preExecute(cmd *cobra.Command, args []string) (err error) {
+	if len(args) != 1 {
+		return fmt.Errorf("incorrect number of args; please specify the puppet version as the only arg")
+	}
+
+	semVer, err := semver.NewVersion(args[0])
```

Comment:
What happens if I specify `7` here? will it try to set to `7.0.0` or the latest 7x? Do we need a handler for less than three digits or should we just document this behavior?

Reply:
I was in the middle of answering this and I'm realising you've raised a good usability conundrum...

So, `semver.NewVersion` will store this as `7.0.0`. We should probably agree on what we do here. My preferred workflow would be:

- `7` means latest version of Puppet 7
- `6` means latest version of Puppet 6
- `7.6` means latest version of `7.6`, so `7.6.1`

...and have all the loveliness of autocompletion too, etc...

But that seems like too much work for a Phase I. We'd need to make a query to each backend, to get all that data, on initialisation. In the case of Docker, it actually doesn't seem like there's a way to do it other than calling the Hub's API.

I think the only option here would be to document this for now - given current designs, I'm not sure we can do anything intelligent on version processing and need to push this on to the Backend, getting back an error if the Backend doesn't have the version in question.

@da-ar thoughts?